staging/src/k8s.io/client-go/tools/leaderelection/leaderelection_test.go

```diff
@@ -1072,3 +1072,128 @@ func TestReleaseLeaseConfigMaps(t *testing.T) {
 func TestReleaseLeaseLeases(t *testing.T) {
 	testReleaseLease(t, "leases")
 }
+
+func TestReleaseOnCancellation_Endpoints(t *testing.T) {
+	testReleaseOnCancellation(t, "endpoints")
+}
+
+func TestReleaseOnCancellation_ConfigMaps(t *testing.T) {
+	testReleaseOnCancellation(t, "configmaps")
+}
+
+func TestReleaseOnCancellation_Leases(t *testing.T) {
+	testReleaseOnCancellation(t, "leases")
+}
+
+func testReleaseOnCancellation(t *testing.T, objectType string) {
+	var (
+		onNewLeader   = make(chan struct{})
+		onRenewCalled = make(chan struct{})
+		onRenewResume = make(chan struct{})
+		onRelease     = make(chan struct{})
+
+		clientMutex sync.RWMutex
```

Comment:
Do you get data races without this lock? These calls should be serialized by the client.

Reply:
I see the earlier races (1.16) are resolved: https://github.com/kubernetes/kubernetes/issues/90952#issuecomment-717664637

Reply:
I got rid of the mutex